src/Compilers/CSharp/Portable/Binder/UsingStatementBinder.cs

```diff
@@ -132,9 +175,27 @@ internal override BoundStatement BindUsingStatementParts(DiagnosticBag diagnosti
                 expressionOpt,
                 iDisposableConversion,
                 boundBody,
+                null, // This ensures that a pattern-matched Dispose statement is not used.
                 hasErrors);
         }
 
+        /// <summary>
+        /// Checks for a Dispose method on exprType. Failing to satisfy the pattern is not an error -
+        /// it just means we have to check for an interface instead.
+        /// </summary>
+        /// <param name="exprType">Type of the expression over which to iterate</param>
+        /// <param name="diagnostics">Populated with warnings if there are near misses</param>
+        /// <returns>True if a matching method is found (still need to verify return type).</returns>
+        private MethodSymbol SatisfiesDisposePattern(TypeSymbol exprType, DiagnosticBag diagnostics)
```

Comment:
I think this method should take the method from `FindPatternMethod` as a parameter and return a `bool` to indicate if it matches. Alternatively, it should probably be renamed `TryFindDisposeMethod`.

Reply:
Can I change the return type to a `bool` and add an `out` parameter for DisposeMethod?